lib/python/Jetson/GPIO/gpio_event.py

```diff
@@ -245,124 +355,52 @@ def _poll_thread():
         finally:
             if _mutex.locked():
                 _mutex.release()
-    thread.exit()
-
-
-def blocking_wait_for_edge(gpio, gpio_name, edge, bouncetime, timeout):
-    global _epoll_fd_blocking
-    gpio_obj = None
-    finished = False
-    res = None
-    initial_edge = True
-
-    if timeout is None:
-        timeout = -1
-    else:
-        timeout = float(timeout) / 1000
-
-    if gpio in _gpio_event_list:
-        if _gpio_event_list[gpio].callbacks:
-            return -1
-
-    # check if gpio edge already added. Add if not already added
-    added_edge = gpio_event_added(gpio)
-
-    # get existing record
-    if added_edge == edge:
-        gpio_obj = _get_gpio_object(gpio)
-        if (gpio_obj.bouncetime is not None and
-                gpio_obj.bouncetime != bouncetime):
-            return -1
-
-    # not added. create new record
-    elif added_edge == NO_EDGE:
-        gpio_obj = _Gpios(gpio_name, edge, bouncetime)
-        _set_edge(gpio_name, edge)
-        _gpio_event_list[gpio] = gpio_obj
-
-    # added_edge != edge. Event is for different edge
-    else:
-        _mutex.acquire()
-        gpio_obj = _get_gpio_object(gpio)
-        _set_edge(gpio_name, edge)
-        gpio_obj.edge = edge
-        gpio_obj.bouncetime = bouncetime
-        gpio_obj.initial_wait = True
-        _gpio_event_list[gpio] = gpio_obj
-        _mutex.release()
 
-    # create epoll blocking object if not already created
-    if _epoll_fd_blocking is None:
-        _epoll_fd_blocking = epoll()
-        if _epoll_fd_blocking is None:
-            return -2
+    thread.exit()
 
-    # register gpio value fd with epoll
+# This function waits for a edge event in a blocking mode, which the user must 
+# specify the file descriptor of the chip, which channel of the chip, the event handle,
+# time for debouncing in milliseconds, the time limit to wait for the event
+def blocking_wait_for_edge(chip_fd, channel, request, bouncetime, timeout):
```

Comment:
It seems like the old function on the left also check whether the gpio is currently being used by edge detects. Can you help me add that in here?